resources/prosody-plugins/mod_polls_component.lua

```diff
@@ -146,11 +146,39 @@ end
             room = get_room_from_jid(room_jid_match_rewrite(json_message.attr.roomJid));
         else
             room = get_room_by_name_and_subdomain(session.jitsi_web_query_room, session.jitsi_web_query_prefix);
-        end
-
-        if not room then
-            module:log('warn', 'No room found found for %s %s', session.jitsi_web_query_room, session.jitsi_web_query_prefix);
-            return;
+            if not room then
+                -- Enhanced logging for debugging room resolution failures
+                module:log('error', '[Polls] No room found for session context: jitsi_web_query_room=%s, jitsi_web_query_prefix=%s, stanza.from=%s, stanza.to=%s',
+                    tostring(session.jitsi_web_query_room), tostring(session.jitsi_web_query_prefix), tostring(stanza.attr.from), tostring(stanza.attr.to));
+                -- Fallback: try to infer room from occupant's current room if in a breakout room context
+                if session.jitsi_web_query_room == nil or session.jitsi_web_query_prefix == nil then
```

Comment:
why would these be nil, they will always be set. Do you see somewhere these to be cleared?